tools/memap.py

```diff
@@ -62,23 +63,69 @@ def __init__(self, detailed_misc=False):
 
         self.subtotal = dict()
 
-    def module_add(self, module_name, size, section):
-        """ Adds a module / section to the list
+        self.misc_flash_mem = 0
+
+
+    def module_remove_unused(self):
+        """ Removes modules/objects that were compiled but are not used
+        """
+
+        # Using keys to be able to remove entry
+        for i in self.modules.keys():
+            size = 0
+            for k in self.print_sections:
+                size += self.modules[i][k]
+            if size == 0:
+                self.modules.pop(i)
+
+
+    def module_init(self, object_name):
+        """ Initialize a module. Just adds the name of the module
 
         Positional arguments:
-        module_name - name of the module to add
-        size - the size of the module being added
-        section - the section the module contributes to
+        object_name - name of the entry to add
         """
 
-        if module_name in self.modules:
-            self.modules[module_name][section] += size
-        else:
+        if object_name not in self.modules:
             temp_dic = dict()
             for section_idx in self.all_sections:
                 temp_dic[section_idx] = 0
-            temp_dic[section] = size
-            self.modules[module_name] = temp_dic
+            self.modules[object_name] = temp_dic
+
+    def module_add(self, object_name, size, section):
+        """ Adds a module / section to the list
+
+        Positional arguments:
+        object_name - name of the entry to add
+        size - the size of the module being added
+        section - the section the module contributes to
+        """
+
+        # Check if object is a sub-string of key
+        for key, value in self.modules.items():
+
+            # this is required to differenciate: main.o vs xxxmain.o
+            key_split = key.split('/')[-1]
+            obj_split = object_name.split('/')[-1]
+
+            if key_split == obj_split:
+                self.modules[key][section] += size
+                return
+
+        temp_dic = dict()
+        for section_idx in self.all_sections:
+            temp_dic[section_idx] = 0
+        temp_dic[section] = size
+        self.modules[object_name] = temp_dic
+
+    def module_replace(self, old_object, new_object):
+        """ Replaces an object name with a new one
+        """
+
+        # Check if object is a sub-string of key
+        if old_object in self.modules:
+            self.modules[new_object] = self.modules[old_object]
+            del self.modules[old_object]
```

Comment:
In `module_remove_unused` we removed objects from the module dict by using the `.pop()` method. Here we use `del`. Could we make these two instances consistent? I don't mind either implementation.